pkg/build/sbom.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"path"
+
+	"istio.io/pkg/log"
+	"istio.io/release-builder/pkg/model"
+	"istio.io/release-builder/pkg/util"
+)
+
+// Sbom generates Software Bill Of Materials for istio repo in an SPDX readable format.
+func GenerateBillOfMaterials(manifest model.Manifest) error {
+	// Retrieve istio repository path to run the sbom generator
+	istioRepoDir := manifest.RepoDir("istio")
+	sourceSbomFile := path.Join(manifest.OutDir(), "istio-source.spdx")
+	sourceSbomNamespace := fmt.Sprintf("https://storage.googleapis.com/istio-release/releases/%s/istio-source.spdx",
+		manifest.Version)
+	releaseSbomFile := path.Join(manifest.OutDir(), "istio-release.spdx")
+	releaseSbomNamespace := fmt.Sprintf("https://storage.googleapis.com/istio-release/releases/%s/istio-release.spdx",
+		manifest.Version)
+
+	// Run bom generator to generate the software bill of materials(SBOM) for istio.
+	log.Infof("Generating Software Bill of Materials for istio release artifacts")
+	if err := util.VerboseCommand("bom", "generate", "--namespace", releaseSbomNamespace,
```

Comment:
I thought the image archives were going to be included here?

Reply:
will include --image-archive once v0.2.0 of bom is released. Currently our build-tools is on 0.1.0. Until then docker image tar balls are also scanned as files.

Reply:
Cool, we are [one PR away](https://github.com/kubernetes-sigs/bom/pull/31) from 0.2, it should be out shortly!

Reply:
@puerco : upgraded to 0.2.0 and using --image-archive now!